docs: add tutorial for setup with k3d and nginx (#3309)
loculus-project/loculus: docs/src/content/docs/for-administrators/setup-with-k3d-and-nginx.mdx

import { Aside } from '@astrojs/starlight/components';

This tutorial shows you how to set up Loculus on a plain Linux server that is reachable through a domain. For a local instance (i.e., an instance that is not publicly accessible), please check out [this tutorial](../my-first-loculus/).

We will use [k3d](https://k3d.io/) to run a Kubernetes cluster within Docker, [nginx](https://nginx.org/) as a proxy server and [Let's Encrypt](https://letsencrypt.org/) to obtain a SSL certificate. The tutorial was tested on a Ubuntu 24.04.

## Prerequisites

-   A Linux server (e.g. Ubuntu 24.04) with ideally at least 4 CPUs and 8 GB RAM
-   A domain
-   Install Docker: see [this guide](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository)
-   Install k3d: see [this guide](https://k3d.io/stable/#installation)
-   Install kubectl: see [this guide](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-kubectl-on-linux)
-   Install Helm: see [this guide](https://helm.sh/docs/intro/install/#from-apt-debianubuntu)
-   Install nginx: see [this guide](https://ubuntu.com/tutorials/install-and-configure-nginx#2-installing-nginx)

## Step 1: Create a k3d cluster

> We prepared a script that creates a k3d cluster, starts Traefik as a proxy server and contains relevant port mappings. Create a file `create-cluster.sh` with the following content, make it executable with `chmod +x create-cluster.sh` and execute it with `./create-cluster.sh`.

Reviewer: Actually it doesn't start Traefik.
```suggestion
We prepared a script that creates a k3d cluster, waits until the Traefik proxy in the cluster is ready and contains relevant port mappings. Create a file `create-cluster.sh` with the following content, make it executable with `chmod +x create-cluster.sh` and execute it with `./create-cluster.sh`.
```

```bash
helm install -f my-values.yaml my-loculus loculus/kubernetes/loculus
```

<Aside>
    To uninstall the helm chart, you can run:

    ```
    helm uninstall my-loculus
    ```

</Aside>

Wait a few moments until everything gets started. You can see the status of the pods with:

```bash
kubectl get pods
```

Once the pods are running, the website is locally available (i.e., from within the sever) at port 9000 and you can download the HTML of the landing page with:

```bash
curl localhost:9000
```

In the output, you should see somewhere:

```
<title>Home | <Your instance name></title>
```

## Step 5: Configure DNS / domain

You have to add DNS entries of your domain to point to the server. Please add DNS entries for the following domains and sub-domains:

-   `<your domain>`
-   `api.<your domain>`
-   `auth.<your domain>`

## Step 6: Set up nginx

Now, you can use nginx as a reverse proxy to make the instance publicly accessible through the domain.

First, delete the default nginx page with:

```bash
sudo rm /etc/nginx/sites-enabled/default
```

> Next, create the config for Loculus at `/etc/nginx/sites-enabled/loculus` with the following content and replace your domain:

Reviewer: ```suggestion
Next, create the config for Loculus at `/etc/nginx/sites-enabled/loculus` with the following content and replace `<your domain>`:
```
